hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/impl/OperationExecutorImpl.java

```diff
@@ -185,6 +203,41 @@ private PartitionOperationThreadImpl[] initPartitionThreads(HazelcastProperties
         return threads;
     }
 
+    private PartitionOperationThread[] initAltoPartitionThreads(TpcEngine tpcEngine,
+                                                                String hzName,
+                                                                NodeExtension nodeExtension,
+                                                                ClassLoader configClassLoader) {
+        int threadCount = tpcEngine.reactorCount();
+
+        PartitionOperationThread[] threads = new PartitionOperationThread[threadCount];
+        for (int threadId = 0; threadId < threads.length; threadId++) {
+            Reactor reactor = tpcEngine.reactor(threadId);
+            AltoPartitionOperationThread partitionThread = (AltoPartitionOperationThread) reactor.eventloopThread();
+
+            partitionThread.setName(createThreadPoolName(hzName, "partition-operation") + threadId);
+
+            // the AltoOperationQueue is unbound just like HZ classic. Since we do not have any back pressure mechanism between
+            // members, there is no proper way to prevent overload. So we keep the same bad bad behavior for now.
+            partitionThread.queue = new AltoOperationQueue(reactor, new MPSCQueue<>(null), new ConcurrentLinkedQueue<>());
```

Comment:
1. Why do we need multi producer queue as normalQueue. Who can add tasks to this queue besides the eventloop thread?
2. Same goes for priorityQueue. Also even if there is multiple producers, why we don't just use MPSC like we did for normalQueue?

Reply:
1: any thread can offer a request for processing. This could be a user thread, but also an I/O thread. So you need to have a MP type of queue.

2: I want to keep semantics as close to the classic threads as possible. So same queues.